### PR TITLE
feat: capsules clawback method

### PR DESCRIPTION
**Pull request by utx0**
## Description

* clawback fund function as part of the `seasonsbase.sol` 
* verifySolvent modifier for easy of use. 
* Updated a bunch of error messages on `coreRef.sol` for consistency
* Added `whenNotpaused` to the redeem function   

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

-   [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [ ] ~added `!` to the type prefix if API or client breaking change~
-   [ ] ~provided a link to the relevant issue or specification~
-   [X] included the necessary unit and integration tests
-   [X] code is adequately commented
-   [ ] ~updated the relevant documentation or specification~
-   [X] reviewed "Files changed" and left comments if necessary
-   [X] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

-   [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [ ] confirmed `!` in the type prefix if API or client breaking change
-   [ ] confirmed all author checklist items have been addressed
-   [ ] reviewed service design and naming
-   [ ] reviewed documentation is accurate
-   [ ] reviewed tests and test coverage
-   [ ] manually tested (if applicable)

### CODE REVIEW

File: src/nfts/seasons/SeasonsBase.sol [hunk 1]
```diff
@@ -20,13 +20,32 @@ struct TokenIdRewardAmount {
...
+
     ERC1155MaxSupplyMintable public immutable nftContract;
     IERC20 public immutable rewardToken;
     SeasonsTokenIdRegistry public immutable tokenIdRegistryContract;
 
     /// @notice Total amount of reward tokens needed by the contract to be solvent
     uint256 public totalRewardTokens;
 
+    /// @notice Total amount of reward tokens clawbacked. Kept for record keeping.
+    uint256 public totalClawbacked;
```
Comment (intl-man) [inline src/nfts/seasons/SeasonsBase.sol:33]: Being pedantic, but should be `totalClawedBack`.

File: test/unit/nfts/seasons/ERC1155SeasonOne.t.sol [hunk 2]
```diff
@@ -203,4 +204,62 @@ contract UnitTestERC1155SeasonOne is SeasonBase {
         assertEq(token.balanceOf(address(this)), 400);
         assertEq(token.balanceOf(address(_seasonOne)), beforeTotalRewardTokens - 400);
     }
+
+    function testRedeemWhenPaused() public {
```
Comment (intl-man) [inline test/unit/nfts/seasons/ERC1155SeasonOne.t.sol:208]: Should also add to the integration tests for pausing the entire system.

**Review by intl-man — CHANGES_REQUESTED** [review]

File: test/unit/nfts/seasons/ERC1155SeasonOne.t.sol (hunk 2 above)
Comment (utx0) [reply test/unit/nfts/seasons/ERC1155SeasonOne.t.sol:208]: yep, new PR

**Review by intl-man — APPROVED** [review]
LGTM.

**Pull request merged**